modules/contact/src/actions/ContactAction.C

```diff
@@ -368,72 +395,161 @@ ContactAction::addMortarContact()
       _problem->addVariable(var_type, variable_name, params);
     };
 
-    // Normal contact: same family/order as primal.
+    // To avoid instabilities in the solution and obtain the full benefits of a variational
+    // enforcement, use of dual mortar is strongly recommended.
```

Comment:
Any reason not to deprecate the `use_dual` parameter and _always_ use dual bases?

Reply:
Some developers may be still making comparisons between dual and standard bases (even if it's just for research purposes). But, from a user point of view, we should do that soon to avoid misuse (note that now, to use mortar, a user should only do `formulation = mortar`). In this code I am deprecating the previous way of doing mortar contact, so we can remove `_use_dual` and the `legacy` approach a few months from now.

Reply:
https://github.com/idaholab/moose/pull/17980#discussion_r643388815